views/js/pciCreator/ims/mathEntryInteraction/creator/widget/states/Map.js

```diff
@@ -188,113 +286,89 @@ define([
+                }
+                this.correctResponses.splice(inputId, 1);
             }
-
-            self.correctResponses.splice(id, 1);
-            self.renderForm(self.correctResponses);
-        });
+        })
     }
 
-    MathEntryInteractionStateResponse.prototype.renderForm = function renderForm(correctAnswerOptions) {
-        var self = this,
-            $responseForm = self.widget.$responseForm;
-
-        this.removeEditDeleteListeners();
-        $responseForm.find('.mathEntryInteraction').remove();
-        $responseForm.append(answerFormTpl({correctAnswerEntries: correctAnswerOptions}));
-        this.initDeletingOptions();
-        this.initEditingOptions();
-    }
 
     /**
      *   remove all event listeners to avoid any potential memory leaks
      */
-    MathEntryInteractionStateResponse.prototype.removeEditDeleteListeners = function removeEditDeleteListeners() {
-        var self = this,
-            $entryConfig = self.widget.$responseForm.find('.entry-config');
-
-        $entryConfig.find('.answer-edit').off('click');
-        $entryConfig.find('.answer-delete').off('click');
+    MathEntryInteractionStateResponse.prototype.removeDeleteListeners = function removeDeleteListeners() {
+        const $deleteButtons = this.widget.$container.find('.entry-config');
+        $deleteButtons.find('.answer-delete').off('click');
     }
 
     MathEntryInteractionStateResponse.prototype.removeAddButtonListener = function removeAddButtonListener() {
-        var self = this,
-            $responseForm = self.widget.$responseForm;
-
-        $responseForm.find($('.add-answer-option')).off('click');
+        $addAlternativeBtn.off('click');
     }
 
     MathEntryInteractionStateResponse.prototype.destroyForm = function destroyForm() {
-        var self = this,
-            $responseForm = self.widget.$responseForm;
-
+        const $responseForm = this.widget.$responseForm;
         $responseForm.find('.mathEntryInteraction').remove();
     }
 
     MathEntryInteractionStateResponse.prototype.saveAnswers = function saveAnswers() {
-        var self = this,
-            interaction = self.widget.element,
-            responseDeclaration = interaction.getResponseDeclaration();
+        const interaction = this.widget.element;
+        const responseDeclaration = interaction.getResponseDeclaration();
 
         this.clearMapEntries();
 
         if (this.inGapMode() === true) {
-            self.correctResponses = self.correctResponses.filter(function (response) {
+            this.correctResponses = this.correctResponses.filter(function (response) {
                 return response.split(',').indexOf('') === -1;
             });
         }
+        const score = [];
+        if (this.correctResponses.length) {
+            const scoreInput = this.widget.$container.find('.math-entry-score-input')
+            scoreInput.map(input => {
+                score.push(scoreInput[input].value);
+            })
```

Comment:
**chore:** What is the point of using `.map()` if you don't expect a return? You should either use `.forEach()` instead or rely on the map to build the new array.

**suggestion:**
- solution 1:
```suggestion
            scoreInput.forEach(input => {
                score.push(scoreInput[input].value);
            });
```
- Solution 2:
```suggestion
            score = scoreInput.map(input => scoreInput[input].value);
```